dimod/constrained.py

```diff
@@ -1056,6 +1056,140 @@ def to_file(self, *, spool_size: int = int(1e9)) -> tempfile.SpooledTemporaryFil
+            f.write("Minimize\n ")
+            obj_string = 'obj: '
+
+            i_term = 0
+            if self.objective.linear:
+                for var, bias in self.objective.linear.items():
+                    obj_string += '{} {} {} '.format(sign_dict[int(np.sign(bias))], abs(bias), var)
+                    i_term += 1
+
+                    if i_term % new_line_every == 0:
+                        obj_string += '\n '
+
+            if self.objective.quadratic:
+                obj_string += '+ [ '
+                for edge, bias in self.objective.quadratic.items():
+                    var1, var2 = edge
+                    obj_string += '{} {} {} * {} '.format(sign_dict[np.sign(bias)], abs(2 * bias), var1, var2)
+                    i_term += 1
+                    if i_term % new_line_every == 0:
+                        obj_string += '\n '
+                obj_string += ']/2 '
+
+            f.write(obj_string)
+            f.write("\n \n")
+
+        if self.constraints:
+            f.write("Subject To \n")
+
+            for label, constraint in self.constraints.items():
+
+                if label[0] in not_first_char:
+                    raise ValueError('Cannot start variable name with {}'.format(label[0]))
+
+                if any([l in not_allowed_chars for l in label]):
+                    raise ValueError('Cannot use characters {} in names'.format(not_allowed_chars))
+
+                c_string = '{}: '.format(label)
+
+                i_term = 0
+                if constraint.lhs.linear:
+                    for var, bias in constraint.lhs.linear.items():
+                        c_string += '{} {} {} '.format(sign_dict[int(np.sign(bias))], abs(bias), var)
+                        i_term += 1
+
+                        if i_term % new_line_every == 0:
+                            c_string += '\n '
+
+                if constraint.lhs.quadratic:
+                    c_string += '+ [ '
+                    for edge, bias in constraint.lhs.quadratic.items():
+                        var1, var2 = edge
+                        c_string += '{} {} {} * {} '.format(sign_dict[np.sign(bias)], abs(bias), var1, var2)
+                        i_term += 1
+                        if i_term % new_line_every == 0:
+                            c_string += '\n '
+                    c_string += '] '
+
+                rhs = constraint.rhs - constraint.lhs.offset
+                c_string += ' {} {} \n'.format(sense_dict[constraint.sense.value], rhs)
+                f.write(c_string)
+
+        vartypes = {var: self.vartype(var) for var in self.variables}
+
+        # write the bounds
+        f.write('\n')
+        f.write('Bounds \n')
+        for v, vartype in vartypes.items():
+            if any([vv in not_allowed_chars for vv in v]):
+                raise ValueError('Cannot use chars {} in variable name'.format(not_allowed_chars))
+            if v[0] in not_first_char:
```

Comment:
`""` and other 0-length objects are allowed as variable labels

Reply:
so since the LP file doesn't allow that how do I check that?

Reply:
Ah, then just raise an informative error in that case.